axoupdater/src/release/github.rs

```diff
@@ -1,17 +1,44 @@
 //! Fetching and processing from GitHub Releases
 
 use super::{Asset, Release};
-use crate::errors::*;
+use crate::{app_name_to_env_var, errors::*};
 use axoasset::reqwest::{
     self,
     header::{ACCEPT, USER_AGENT},
 };
 use axotag::{parse_tag, Version};
 use serde::{Deserialize, Serialize};
 use std::env;
+use url::Url;
+
+fn github_api(app_name: &str) -> AxoupdateResult<String> {
+    let formatted_app_name = app_name_to_env_var(app_name);
+    let ghe_env_var = format!("{}_INSTALLER_GHE_BASE_URL", formatted_app_name);
+    let github_env_var = format!("{}_INSTALLER_GITHUB_BASE_URL", formatted_app_name);
+
+    if env::var(&ghe_env_var).is_ok() && env::var(&github_env_var).is_ok() {
+        return Err(AxoupdateError::MultipleGitHubAPIs {
+            ghe_env_var,
+            github_env_var,
+        });
+    }
 
-fn github_api() -> String {
-    env::var("INSTALLER_BASE_URL").unwrap_or_else(|_| "https://api.github.com".to_string())
+    if let Ok(value) = env::var(&ghe_env_var) {
```

Comment:
spiritually i feel like this wants to be a match not an if since we don't allow the 2 conditions

Reply:
I don't see how to write this code as a match! I think it's fine as-is